### PR TITLE
Documentation for std::invoke

### PR DESCRIPTION
**Pull request by alexandraepan**
### Description 
Create a new file with the description of std::invoke. The output available at https://alexandraepan.github.io/oneTBB/main/tbb_userguide/std_invoke.html 

Fixes # - _issue number(s) if exists_

- [x] - git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/oneapi-src/oneTBB/blob/master/CONTRIBUTING.md#pull-requests) for details)_

### Type of change

_Choose one or multiple, leave empty if none of the other choices apply_

_Add a respective label(s) to PR if you have permissions_

- [ ] bug fix - _change that fixes an issue_
- [ ] new feature - _change that adds functionality_
- [ ] tests - _change in tests_
- [ ] infrastructure - _change in infrastructure and CI_
- [x] documentation - _documentation update_

### Tests

- [ ] added - _required for new features and some bug fixes_
- [x] not needed

### Documentation

- [ ] updated in # - _add PR number_
- [ ] needs to be updated
- [x] not needed

### Breaks backward compatibility
- [ ] Yes
- [x] No
- [ ] Unknown

### Notify the following users
_List users with `@` to send notifications_

### CODE REVIEW

File: doc/main/tbb_userguide/std_invoke.rst [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+==========================
+
+Starting from C++17, the requirements for callable objects passed to algorithms or Flow Graph nodes are relaxed. It allows using additional types of bodies. 
+Previously, the body of the algorithm or Flow Graph node needed to be a Function Object (see `C++ Standard Function Object <https://en.cppreference.com/w/cpp/utility/functional>`_) and provide an 
+``operator()`` that accepts input parameters. 
+
+Now the body needs to meet the more relaxed requirements of being Callable (see `C++ Standard Callable <https://en.cppreference.com/w/cpp/named_req/Callable>`_) that covers three types of objects:
+
+* **Function Objects that provide operator(arg1, arg2, ...)**, which accepts the input parameters
+* **Pointers to member functions** that you can use as the body of the algorithm or parallel construct
```
Comment (kboyarinov) [inline doc/main/tbb_userguide/std_invoke.rst:13]: For me, it is not clear what "parallel construct" is in this context? May be we can use Flow Graph node directly instead?

File: doc/main/tbb_userguide/std_invoke.rst [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+       delete[] array_to_iterate;
+   }
+
+``std::invoke``
+****************
+
+``std::invoke`` is a function template that provides a syntax for invoking different types of callable objects with a set of arguments.
+
+oneTBB implementation uses the C++ standard function ``std::invoke(&StrideRange::iterate, range) <std::invoke link>`` to execute the body. It is the equivalent of ``range.iterate()``.
```
Comment (kboyarinov) [inline doc/main/tbb_userguide/std_invoke.rst:123]:
```suggestion
oneTBB implementation uses the C++ standard function ``std::invoke(&StrideRange::iterate, range)`` to execute the body. It is the equivalent of ``range.iterate()``.
```

File: doc/main/tbb_userguide/std_invoke.rst [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+       make_edge(func1, func2);
+       
+       // Connect the function nodes
+       func1.try_put(Object
+       {
+           1 });
+       // Wait for the graph to complete
+       g.wait_for_all();
+
+   }
```
Comment (pavelkumbrasev) [inline doc/main/tbb_userguide/std_invoke.rst:220]:
```suggestion
    struct Object {
        int number;
    };

    int main() {
        using namespace oneapi::tbb::flow;
 
        // The processing logic for the received integer
        auto number_processor = [] (int i) { /* processing integer */ };

        // Create a graph object g to hold the flow graph
        graph g;

        // Use a member function pointer to the number member of the Object struct as the body
        function_node<Object, int> func1(g, unlimited, &Object::number);

        // Use the number_processor lambda function as the body
        function_node<int, int> func2(g, unlimited, number_processor);

        // Connect the function nodes
        make_edge(func1, func2);

        // Connect the function nodes
        func1.try_put(Object{1});

       // Wait for the graph to complete
       g.wait_for_all();
    }
```

**Review by pavelkumbrasev — APPROVED** [review]

**Review by omalyshe — APPROVED** [review]
LGTM

**Pull request merged**